lib/repl.js

```diff
@@ -282,17 +282,24 @@ function REPLServer(prompt,
   self._domain.on('error', function debugDomainError(e) {
     debug('domain error');
     const top = replMap.get(self);
+    var out;
     internalUtil.decorateErrorStack(e);
-    if (e instanceof SyntaxError && e.stack) {
+
+    if (!((e instanceof Error) && typeof e.stack === 'string')) {
+      out = e;
+    } else if (e instanceof SyntaxError) {
```

Comment:
if there a chance `e.stack != true || !('replace' in e.stack)`?
Maybe a custom error:
```
function MySillyError(a, b, c) { console.log(a) }
MySillyError.prototype = SyntaxError
```